fairseq/trainer.py

```diff
@@ -472,18 +472,17 @@ def maybe_no_sync():
             # normalize grads by sample size
             if sample_size > 0:
                 self.optimizer.multiply_grads(self.args.distributed_world_size / float(sample_size))
+            # tpu-comment: for tpu training, we need to explicitly reduce
+            #   gradients here
+            xm.reduce_gradients(optimizer)
```

Comment:
Before there was some XLA-conditional code.
now you assume here is always XLA.
Is that OK?

Reply:
urgh good catch. Thanks. I struggle with this all the time.